prow/cmd/peribolos/main.go

```diff
@@ -908,17 +967,38 @@ func configureRepos(client repoClient, orgName string, orgConfig org.Config) err
 		byName[strings.ToLower(repo.Name)] = repo
 	}
 
-	var createErrors []error
+	var allErrors []error
 	for wantName, wantRepo := range orgConfig.Repos {
 		if _, have := byName[strings.ToLower(wantName)]; !have {
 			logrus.WithField("repo", wantName).Info("repo does not exist, creating")
-			if _, err := client.CreateRepo(orgName, false, newRepoCreateRequest(wantName, wantRepo)); err != nil {
-				createErrors = append(createErrors, err)
+			created, err := client.CreateRepo(orgName, false, newRepoCreateRequest(wantName, wantRepo))
```

Comment:
Correct. Or set a `default_branch` on it, which unfortunately fails, because the repo may not have any git content yet. It is not ideal.

The question is, how to handle this situation better? If peribolos won't do these `UpdateRepo` calls in the same run, it will do them in the next run anyway (I assume peribolos is intended to be run regularly & often). Maybe it should error out: I usually prefer tools to not be opinionated about what is supposed to be a valid operation, but if desired, I can add check for this.

Reply:
I would say that if the repo is to be archived and doesn't yet exist then don't create it